app/assets/javascripts/visualizations/highvis/scatter.coffee

```diff
@@ -577,177 +492,150 @@ $ ->
-          """
-          </select></td></tr>
-          </table>
-          <button id='regressionButton' class='save_button btn btn-default'>Draw Best Fit Line</button>
-          <table id='regression-table'>
-          <col width='55%' />
-          <col width='35%' />
-          <col width='10%' />
-          <tr id='regression-table-header'><td><strong>f(x)<strong></td><td><strong>Type<strong></td></tr>
-          <tbody id='regression-table-body'></tbody></table>
-          </div></div>
-          """
-
-        # Write HTML
-        $('#controldiv').append controls
-        $("#regressionControl button").button()
-
-        # Add all the saved regressions correctly
-        for regression in @configs.savedRegressions
-          # Filter out the ones that should be enabled.
-          # - X indices must match.
-          # - Compare the arrays without comparing them.
-          # - Y axis must be present.
-          if regression.xAxis == @configs.xAxis \
-          and "#{regression.groups}" is "#{data.groupSelection}" \
-          and globals.configs.fieldSelection.indexOf(regression.yAxis) != -1
-            # Calculate the hypothesis function
-            func = if regression.type is globals.REGRESSION.SYMBOLIC
-              new Function("x", regression.func)
-            else
-              new Function("x, P", regression.func)
-            # Calculate the series
-            series = if regression.type isnt globals.REGRESSION.SYMBOLIC
-              # Convert parameters from strings to numbers
-              params = for i in [0...regression.parameters.length]
-                parseFloat(regression.parameters[i])
-              globals.getRegressionSeries(func, params, \
-              Number(regression.r2), regression.type, [@configs.xBounds.min, @configs.xBounds.max], \
-              regression.name, regression.dashStyle, regression.id, regression.tooltip, false)[3]
+        regressions = ['Linear', 'Quadratic', 'Cubic', 'Exponential',
+            'Logarithmic', 'Automatic']
+
+        ictx =
+          xAxis: data.fields[@configs.xAxis].fieldName
+          yFields:
+            for f in globals.configs.fieldSelection
+              name: data.fields[f].fieldName
+              id: f
+          regressions: regressions
+
+        octx = {}
+        octx.id = 'regression-ctrls'
+        octx.title = 'Analysis Tools'
+        octx.body = HandlebarsTemplates[hbCtrl('regr')](ictx)
+        tools = HandlebarsTemplates[hbCtrl('body')](octx)
+        $('#vis-ctrls').append tools
+
+        # Initialize and track the status of this control panel
+        globals.configs.regressionOpen ?= false
+        initCtrlPanel('regression-ctrls', 'regressionOpen')
+
+        # Restore saved regressions if they exist
+        fs = globals.configs.fieldSelection
+        for regr in @configs.savedRegressions
+          groupsIntersect =
+            (g in data.groupSelection for g in regr.groups).reduce(
+              ((p, c) -> p and c), true)
```

Comment:
This intersection code is repeated in a couple places - should we introduce a function ?